Empire/InfluenceQueue.cpp

```diff
@@ -0,0 +1,208 @@
+{ return !m_queue.size(); }
+
+unsigned int InfluenceQueue::size() const
+{ return m_queue.size(); }
+
+InfluenceQueue::const_iterator InfluenceQueue::begin() const
+{ return m_queue.begin(); }
+
+InfluenceQueue::const_iterator InfluenceQueue::end() const
+{ return m_queue.end(); }
+
+InfluenceQueue::const_iterator InfluenceQueue::find(const std::string& item_name) const {
+    for (auto it = begin(); it != end(); ++it) {
+        if (it->name == item_name)
+            return it;
+    }
+    return end();
+}
+
+const InfluenceQueue::Element& InfluenceQueue::operator[](int i) const {
+    assert(0 <= i && i < static_cast<int>(m_queue.size()));
+    return m_queue[i];
+}
+
+void InfluenceQueue::Update() {
+    const Empire* empire = GetEmpire(m_empire_id);
+    if (!empire) {
+        ErrorLogger() << "InfluenceQueue::Update passed null empire.  doing nothing.";
+        m_projects_in_progress = 0;
+        return;
+    }
+
+    ScopedTimer update_timer("InfluenceQueue::Update");
+
+    float available_IP = empire->ResourceOutput(RE_INFLUENCE);
+    float stockpiled_IP = empire->ResourceStockpile(RE_INFLUENCE);
+
+    float spending_on_policy_adoption_ip = 0.0f;
+    for (auto policy_turn : empire->TurnsPoliciesAdopted()) {
+        if (policy_turn.second != CurrentTurn())
+            continue;
+        auto policy = GetPolicy(policy_turn.first);
+        if (!policy) {
+            ErrorLogger() << "InfluenceQueue::Update couldn't get policy supposedly adopted this turn: " << policy_turn.first;
+            continue;
+        }
+        spending_on_policy_adoption_ip += policy->AdoptionCost(m_empire_id);
+    }
+
+    m_total_IPs_spent = spending_on_policy_adoption_ip;
+
+    m_expected_new_stockpile_amount = stockpiled_IP + available_IP - m_total_IPs_spent;
+
+    DebugLogger() << "InfluenceQueue::Update : available IP: " << available_IP << "  stockpiled: " << stockpiled_IP << "  new expected: " << m_expected_new_stockpile_amount << std::endl;
+
+    //// cache Influence item costs and times
+    //// initialize Influence queue item completion status to 'never'
+    //boost::posix_time::ptime sim_time_start;
+    //boost::posix_time::ptime sim_time_end;
+
+    //DebugLogger() << "InfluenceQueue::Update: Projections took "
+    //              << ((sim_time_end - sim_time_start).total_microseconds()) << " microseconds with "
+    //              << empire->ResourceOutput(RE_INFLUENCE) << " influence output";
+    InfluenceQueueChangedSignal();
+}
+
+void InfluenceQueue::push_back(const Element& element)
+{ m_queue.push_back(element); }
+
+void InfluenceQueue::insert(iterator it, const Element& element)
+{ m_queue.insert(it, element); }
+
+void InfluenceQueue::erase(int i) {
+    assert(i <= static_cast<int>(size()));
+    m_queue.erase(begin() + i);
+}
+
+InfluenceQueue::iterator InfluenceQueue::erase(iterator it) {
+    assert(it != end());
+    return m_queue.erase(it);
+}
+
+InfluenceQueue::iterator InfluenceQueue::begin()
```

Comment:
Trailing type return value would remove the need to repeat the class name:

```suggestion
auto InfluenceQueue::begin() -> iterator
```